### DISCUSSION TITLE
Text viewer must not allow selection edge inside Complex emoji

### DISCUSSION THREAD

**Discussion by Alexey-T** [doublecmd/doublecmd#1091] [Suggestions]
### Description
Linux gtk2. Ubuntu 20.x x64.

Text file
```
abc
def
    👨🏻‍🦰 an: text
```

Make mouse selection from begin until emoji position. viewer now allows mouse position INSIDE the emoji, so partial emoji is rendered when you select by mouse.
I suggest to disallow mouse position inside Complex emoji.

**Comment by Alexey-T** [comment]
I add that emoji “red man” consists of 3 chars -

- man
- Joining char
- Red hairs

**Comment by rich2014** [comment]
it's a bug about CodePoint/CodeUnit of SynEdit.

PS. i recall AtSynEdit had a similar problem.

there is a plan to fix this after Lazarus IME MR 40008 was merged
(https://gitlab.com/freepascal.org/lazarus/lazarus/-/merge_requests/121)
